### PR TITLE
Make pulse dependency optional

### PR DESCRIPTION
**Pull request by soreau**
This makes building gvc and volume widget optional which is needed
to build wf-shell without pulseaudio dependency.

### CODE REVIEW

**Comment by soreau** [comment]
Yea, I think I can do it with a single line like [this](https://github.com/WayfireWM/wf-shell/blob/master/meson.build#L30). I can't recall a specific reason for adding a config.h file.

File: src/panel/panel.cpp [hunk 1]
```diff
@@ -201,8 +203,10 @@ class WayfirePanel::impl
             return Widget(new WayfireNetworkInfo());
         if (name == "battery")
             return Widget(new WayfireBatteryInfo());
+#ifdef HAVE_PULSE
         if (name == "volume")
```
Comment (ammen99) [inline src/panel/panel.cpp]: I'd also add an `#else` to print a warning if volume was disabled (otherwise it might be confusing for the user)

Comment (soreau) [reply src/panel/panel.cpp]: Added.

Comment (ammen99) [reply src/panel/panel.cpp]: I meant runtime warning if the volume widget is specified in the config file :)

**Comment by ammen99** [comment]
LGTM.

**Pull request merged**